corehq/apps/app_manager/models.py

```diff
@@ -4521,22 +4422,52 @@ def make_build(self, comment=None, user_id=None):
         # I'm putting this assert here if copy._id is ever None
         # which makes tests error
         assert copy._id
+        prune_auto_generated_builds.delay(self.domain, self._id)
+
+        return copy
 
+    def convert_app_to_build(self, copy_of, user_id, comment=None):
+        self.copy_of = copy_of
         built_on = datetime.datetime.utcnow()
-        copy.date_created = built_on
-        copy.built_on = built_on
-        copy.built_with = BuildRecord(
-            version=copy.build_spec.version,
-            build_number=copy.version,
+        self.date_created = built_on
+        self.built_on = built_on
+        self.built_with = BuildRecord(
+            version=self.build_spec.version,
+            build_number=self.version,
             datetime=built_on,
         )
-        copy.build_comment = comment
-        copy.comment_from = user_id
-        copy.is_released = False
+        self.build_comment = comment
+        self.comment_from = user_id
+        self.is_released = False
+
+    def convert_build_to_app(self):
```

Comment:
I really like the framing of `convert_app_to_build` and vice-versa, but I do notice that `_meta_fields` is used in `export_json` as a list of fields to be excluded.  I think it implements the same concept as `convert_build_to_app`.  If that's true, it'd be good to combine the two (maybe do the `_meta_fields` stuff here in `convert_build_to_app`, say)

Reply:
I don't think these are quite the same pieces of functionality, given that `VersionedDoc.make_build` excludes a similar but not identical list of properties: https://github.com/dimagi/commcare-hq/blob/dcaeefa1dd579d43b1eabe6795216c555f3d2eb5/corehq/apps/app_manager/models.py#L3912

Reply:
Yeah, I see that as well.  Seems like there's some overlap between `convert_build_to_app`, `export_json`, and `from_source`, at least conceptually, but I do see there are differences too.  Not sure whether the differences are intentional/desirable or accidental.  I'll leave it to your judgement.